### ISSUE TITLE
.Net 8 Preview sending email exception

### ISSUE THREAD

**Issue by CuteLeon** [dotnet/runtime#91377]
### Description

Met exception when sending email on .Net 8.

### Reproduction Steps

``` csharp
using var mailMessage = new MailMessage();
mailMessage.From = new MailAddress("systems@xxxxxxx.com.cn", "FromName");
mailMessage.To.Add("receiver@xxxxxx.com.cn");
mailMessage.Subject = "Test Subject";
mailMessage.Body = "Test Body";

ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
using var smtpClient = new SmtpClient("smtp.office365.com", 587);
smtpClient.Credentials = new NetworkCredential("systems@xxxxxx.com.cn", "Password");
smtpClient.DeliveryMethod = SmtpDeliveryMethod.Network;
smtpClient.EnableSsl = true;
smtpClient.Timeout = 5000;
smtpClient.SendCompleted += (sender, e) =>
{
    Console.WriteLine(e.Error?.Message ?? string.Empty);
};
await smtpClient.SendMailAsync(mailMessage);

Console.ReadLine();
```

### Expected behavior

Send email successfully.

### Actual behavior

Exception thrown from `await smtpClient.SendMailAsync(mailMessage);`

> System.Net.Mail.SmtpException: 'The server returned an invalid response to the EHLO command.'

### Other information

Same code works well on .Net 7 and .Net 6, but never work on .Net 8.

**Comment by MihaZupan** [comment]
I can reproduce the issue - I'm able to send mail with the provided repro on 6.0 and 7.0, but hit the exception on 8.0.

**Comment by F1schk0pp** [comment]
Same here. In .NET 6.0 and 7.0 it works but with 8.0 i got the same exception.

**Comment by devFrame** [comment]
getting an EHLO from the other side too :)

**Comment by CuteLeon** [comment]
Hi @MihaZupan, is there any progress or plan on this issue?

**Comment by ManickaP** [comment]
@CuteLeon I'm looking at the issue, I was able to reproduce as well. I'll let you know here when I find out the root cause.

**Comment by CuteLeon** [comment]
> @CuteLeon I'm looking at the issue, I was able to reproduce as well. I'll let you know here when I find out the root cause.

That's fantastic and thank you very much.

**Comment by karelz** [comment]
Reopening to track the backport to 8.0 (RC2).

**Comment by F1schk0pp** [comment]
Thank you very much @ManickaP :)

**Comment by karelz** [comment]
Fixed in main (9.0) in PR #91750 and in 8.0 (RC2) in PR #91794
